tests/lib/tools/boot-state

```diff
@@ -0,0 +1,203 @@
+#!/bin/bash
+
+show_help() {
+    echo "usage: bootenv show [var]"
+    echo "       bootenv set <var> <value>"
+    echo "       bootenv unset <var>"
+    echo "       boot-path"
+    echo "       wait-core-post-boot"
+    echo ""
+    echo "Get information and manages the boot loader for the current system"
+    echo ""
+    echo "COMMANDS:"
+    echo "  bootenv show:         prints the whole bootenv or just the variable passed as parameter"
+    echo "  bootenv set:          sets the given var and value on boot configuration"
+    echo "  bootenv unset:        unsets the given var from boot configuration"
+    echo "  boot-path:            prints the boot path"
+    echo "  wait-core-post-boot:  waits until the snap_mode bootenv var is empty"
+}
+
+get_grub_editenv() {
+    if [ -e /boot/grub2/grubenv ]; then
+        command -v grub2-editenv
+    elif [ -e /boot/grub/grubenv ]; then
+        command -v grub-editenv
+    fi
+}
+
+run_grub_editenv() {
+    local CMD=$(get_grub_editenv)
+
+    if [ -n "$CMD" ]; then
+        "$CMD" "$@"
+    else
+        return 1
+    fi
+}
+
+get_grub_envfile() {
+    if [ -e /boot/grub2/grubenv ]; then
+        echo /boot/grub2/grubenv
+    elif [ -e /boot/grub/grubenv ]; then
+        echo /boot/grub/grubenv
+    else
+        echo ""
+    fi
+}
+
+bootenv() {
+    case "${1:-}" in
+        show)
+            shift
+            bootenv_show "$@"
+            exit
+            ;;
+        set)
+            shift
+            bootenv_set "$@"
+            exit
+            ;;
+        unset)
+            shift
+            bootenv_unset "$@"
+            exit
+            ;;
+        *)
+            echo "boot-state: unsupported bootenv subcommand $1" >&2
+            show_help
+            exit 1
+            ;;
+    esac
+}
+
+bootenv_show() {
+    local var="${1:-}"
+    local GRUBENV_FILE
```

Comment:
Nitpick, you can use lower-case names for local variables. I think everyone uses upper-case variables in shell because they are otherwise global and it sticks out more.

Reply:
sure, thanks for the suggestion